Mathlib/RingTheory/PowerSeries/Inverse.lean

```diff
@@ -236,6 +307,84 @@ instance : LocalRing R⟦X⟧ :=
 
 end LocalRing
 
+section DiscreteValuationRing
+
+variable {k : Type*} [Field k]
+
+open DiscreteValuationRing
+
+theorem hasUnitMulPowIrreducibleFactorization :
+    HasUnitMulPowIrreducibleFactorization k⟦X⟧ :=
+  ⟨X, And.intro X_irreducible
+      (by
+        intro f hf
+        use f.order.get (order_finite_iff_ne_zero.mpr hf)
+        use Unit_of_divided_by_X_pow_order f
+        simp only [Unit_of_divided_by_X_pow_order_nonzero hf]
+        exact self_eq_X_pow_order_mul_divided_by_X_pow_order hf)⟩
+
+instance : UniqueFactorizationMonoid k⟦X⟧ :=
+  hasUnitMulPowIrreducibleFactorization.toUniqueFactorizationMonoid
+
+instance : DiscreteValuationRing k⟦X⟧ :=
+  ofHasUnitMulPowIrreducibleFactorization hasUnitMulPowIrreducibleFactorization
+
+instance isNoetherianRing : IsNoetherianRing k⟦X⟧ :=
+  PrincipalIdealRing.isNoetherianRing
+
+-- The maximal ideal of `k⟦X⟧` is generated by `X`
+theorem maximalIdeal_eq_span_X : LocalRing.maximalIdeal (k⟦X⟧) = Ideal.span {X} := by
+  have hX : (Ideal.span {(X : k⟦X⟧)}).IsMaximal := by
+    rw [Ideal.isMaximal_iff]
+    constructor
+    · rw [Ideal.mem_span_singleton]
+      exact Prime.not_dvd_one X_prime
+    · intro I f hI hfX hfI
+      rw [Ideal.mem_span_singleton, X_dvd_iff] at hfX
+      have hfI0 : C k (f 0) ∈ I := by
+        have : C k (f 0) = f - (f - C k (f 0)) := by rw [sub_sub_cancel]
+        rw [this]
+        apply Ideal.sub_mem I hfI
+        apply hI
+        rw [Ideal.mem_span_singleton, X_dvd_iff, map_sub, constantCoeff_C, ←
+          coeff_zero_eq_constantCoeff_apply, sub_eq_zero, coeff_zero_eq_constantCoeff]
+        rfl
+      rw [← Ideal.eq_top_iff_one]
+      apply Ideal.eq_top_of_isUnit_mem I hfI0 (IsUnit.map (C k) (Ne.isUnit hfX))
+  rw [LocalRing.eq_maximalIdeal hX]
+
+instance isDedekindDomain : IsDedekindDomain k⟦X⟧ :=
```

Comment:
Isn't this found automatically?

Reply:
Yes, removed (it was not the case in Lean3)